crates/bench/src/schemas.rs

```diff
@@ -179,6 +183,34 @@ pub fn create_sequential<T: RandomTable>(seed: u64, count: u32, buckets: u64) ->
     (0..count).map(|id| T::gen(id, &mut rng, buckets)).collect()
 }
 
+/// Create a table whose first `identical` rows are identical except for their `id` column.
+/// The remainder of the rows in the table are random. The total size of the table is `total`.
+/// Intended for filter benchmarks.
+pub fn create_partly_identical<T: RandomTable>(seed: u64, identical: u64, total: u64) -> Vec<T> {
+    // large to avoid duplicates.
+    // technically, overlaps with the identical part of the table can still occur;
+    // in a given row+column, this happens with probability 1 / 2^32, which is negligible.
+    // We use u32::max because sqlite sometimes chokes on large u64s.
+    let buckets = u32::MAX as u64;
+
+    let mut rng = XorShiftLite(seed);
+    let mut result = Vec::with_capacity(total as usize);
+    let mut id = 0;
+    for _ in 0..identical {
+        // clone to preserve rng state
+        let mut rng_ = rng.clone();
+        result.push(T::gen(id as u32, &mut rng_, buckets));
+        id += 1;
+    }
+    // advance rng
+    drop(T::gen(id as u32, &mut rng, buckets));
```

Comment:
Another idiom question for me, is just `let _ = T::gen(...)` more idiomatic here than `drop`? Maybe for @coolreader18